pkg/R/deserialize.R

```diff
@@ -122,21 +123,24 @@ readRowList <- function(obj) {
   # the numCols bytes inside the read function in order to correctly
   # deserialize the row.
   rawObj <- rawConnection(obj, "r+")
-  numCols <- SparkR:::readInt(rawObj)
-  rowOut <- SparkR:::readRow(rawObj, numCols)
-  close(rawObj)
-  rowOut
+  on.exit(close(rawObj))
```

Comment:
I didn't get it, why it will try to reuse rawConnection? I see `on.exit` as something as `finally {}` in Java.